Tests/scripts/collect_tests/collect_tests.py

```diff
@@ -466,6 +482,57 @@ def _collect_pack(
 
         return CollectionResult(
             test=None,
+            mr_to_test=None,
+            pack=pack_id,
+            reason=reason,
+            version_range=version_range,
+            reason_description=reason_description,
+            conf=self.conf,
+            id_set=self.id_set,
+            is_nightly=is_nightly
+        )
+
+    def _collect_pack_for_modeling_rule(
+        self, pack_id: str, reason_description: str, changed_file_path: Path,
+        content_item_range: Optional[VersionRange] = None, is_nightly: bool = False,
+        reason: Optional[CollectionReason] = None
+    ) -> CollectionResult:
+        """Create a CollectionResult for a pack because of a modeling rule
+
+        Marks the pack being collected and the modeling rule that needs to be tested
+
+        Args:
+            pack_id (str): the id of the pack being collected
+            reason (Optional[CollectionReason]): the reason the pack is being collected. Defaults to None.
+            reason_description (str): the reason the pack is being collected
+            changed_file_path (Path): the path to the file that was modified
+            content_item_range (Optional[VersionRange], optional): version range. Defaults to None.
+            is_nightly (Optional[bool]): whether this is a nightly flow. Defaults to False.
+
+        Returns:
+            CollectionResult: the object detailing the pack to collect and the modeling rule that should be tested
+        """
+        pack = PACK_MANAGER.get_pack_metadata(pack_id)
+
+        version_range = content_item_range \
+            if pack.version_range.is_default \
+            else (pack.version_range | content_item_range)
+
+        if not reason:
+            reason = CollectionReason.MODELING_RULE_CHANGED
+            file_type = find_type(changed_file_path.as_posix())
+            if file_type == FileType.MODELING_RULE_SCHEMA:
```

Comment:
Maybe add `else` for default, with a message saying "`file_type not supported`", or something?

Reply:
But we should always have a reason for collecting the Modeling Rule - it doesn't make sense to collect it unless we have an explicit reason to.